lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -591,11 +591,35 @@ def enroll_in_program(cls, input_data):
         # Get the program using the code
         program = get_object_or_404(Program, program_code=program_code)
 
+        # Check if program is a specialisation and get its "parent" program
+        parent_program = None
+        if program.specialization_for:
+            try:
+                parent_program = Program.objects.get(program_code__iexact=program.specialization_for)
+            except ObjectDoesNotExist:
+                log.exception("**Parent program not found: %s**", program.specialization_for)
```

Comment:
Are we expecting these links to be broken?

Reply:
Not per se, it is a defensive feature in case the field is populated with an erroneous/non-existing program code - which is not impossible since it is a free text field.